internal/controller/resource_manager.go

```diff
@@ -116,7 +145,7 @@ func (r *ResourceManager) DynamicListResources(ctx context.Context, resource str
 // watchEvents watches Kubernetes resources and updates cache based on events.
 // Any occurring errors are sent through errChanWatch. The watch stops when ctx is cancelled.
 func (r *ResourceManager) watchEvents(ctx context.Context, resource string, watchOptions metav1.ListOptions, c CacheManager) error {
-	objGVR := schema.GroupVersionResource{Group: "", Version: "v1", Resource: resource}
+	objGVR := schema.GroupVersionResource{Group: resourceAPIGroupMap[resource], Version: "v1", Resource: resource}
```

Comment:
Instead of doing that, why not pass both the API group and resource kind as parameters? You could also define a small struct type with that pair of fields, and pass that instead of just the resource kind.